cli/internal/transformerpipeline/pipeline.go

```diff
@@ -70,31 +71,49 @@ func (lp *TransformerPipeline) Send(data []byte) error {
 	}
 
 	if lp.clientWrappers[0].isClosed {
-		return errors.New("cannot send data to a closed pipeline")
+		return nil
 	}
 
-	return lp.clientWrappers[0].client.Send(&plugin.Transform_Request{Record: data})
+	sendCh := make(chan error)
+
+	// Send can block forever (e.g. if grpc buffer is full), so we run it asynchronously
+	// and check if pipeline is closed every second.
+	go func() {
```

Comment:
What is the benefit of explicitly closing this channel? Btw, this goroutine is never explicitly exited either; the `defer` might as well be outside of it.

Reply:
Is the `client.Send()` method blocking ad infinitum (provided no error was encountered?) I thought (well, the comment above says...) it would only block if grpc buffer is full, or such cases.

Reply:
Closing would be to stop having unused open channels (do they use file descriptor limits? probably not, but don't know)

Reply:
The case that we're hitting is that:
- the source (via `n` workers) is sending non-stop
- the destination was fine but then errors
- the `send` doesn't get `recv`'d many times in a row until the buffer is full

But with my fix, within <= 1 second, the pipeline is closed.

This should be a single channel that is needlessly around for about 1 sec, so I think we're fine.